src/compose/types/record.test.ts

```diff
@@ -204,4 +204,33 @@ describe(__filename, () => {
         '[{"name":"simple","value":"foo"},{"name":"multi","value":"bar"},{"name":"multi","value":"baz"}]')
     })
   })
+
+  describe('prevent value corruption when when value prop is missing on rawValue and one of the fields is named "name"', () => {
+    const m = Object.freeze(new Module({
+      fields: [
+        { name: 'name' },
+        { name: 'alt' },
+      ],
+    }))
+
+    it('it should not set value for name when constructing an object', () => {
+      const r = new Record(m, { name: 'alt' })
+      expect(r.values.name).to.eq(undefined)
+      expect(r.values.alt).to.eq(undefined)
+    })
+
+    it('it should not set value for name when applying new values', () => {
+      const r = new Record(m)
+      r.apply({ name: 'alt' })
+      expect(r.values.name).to.eq(undefined)
+      expect(r.values.alt).to.eq(undefined)
+    })
```

Comment:
@darh shouldn't these ones **set** the value of field `name` to `'alt'`, since we are allowing values to be passed in as a plain object.